include/supercluster.hpp

```diff
@@ -22,22 +22,22 @@ namespace supercluster {
 using namespace mapbox::geometry;
 
 struct Cluster {
-    point<double> pos;
-    std::uint32_t num_points;
+    const point<double> pos;
+    const std::uint32_t num_points;
     std::uint32_t id;
     std::uint32_t parent_id = 0;
     bool visited = false;
 
-    Cluster(point<double> pos_, std::uint32_t num_points_, std::uint32_t id_)
+    Cluster(const point<double> pos_, const std::uint32_t num_points_, const std::uint32_t id_)
         : pos(std::move(pos_)), num_points(num_points_), id(id_) {
```

Comment:
if you make the point `const`, then you can't move it anymore. Given that a point only contains scalar values, `move` doesn't have any benefit anyway. Alternatively, you can change the signature to `const point<double>& pos_`, since we're creating a copy anyway.